cmd/influx/task.go

```diff
@@ -3,342 +3,425 @@ package main
-		id, err := influxdb.IDFromString(taskFindFlags.user)
+	if t.taskFindFlags.user != "" {
+		id, err := influxdb.IDFromString(t.taskFindFlags.user)
 		if err != nil {
 			return err
 		}
 		filter.User = id
 	}
 
-	if taskFindFlags.org.name != "" {
-		filter.Organization = taskFindFlags.org.name
+	if t.org.name != "" {
+		filter.Organization = t.org.name
 	}
-	if taskFindFlags.org.id != "" {
-		id, err := influxdb.IDFromString(taskFindFlags.org.id)
+	if t.org.id != "" {
+		id, err := influxdb.IDFromString(t.org.id)
 		if err != nil {
 			return err
 		}
 		filter.OrganizationID = id
 	}
 
-	if taskFindFlags.limit < 1 || taskFindFlags.limit > influxdb.TaskMaxPageSize {
+	if t.taskFindFlags.limit < 1 || t.taskFindFlags.limit > influxdb.TaskMaxPageSize {
 		return fmt.Errorf("limit must be between 1 and %d", influxdb.TaskMaxPageSize)
 	}
-	filter.Limit = taskFindFlags.limit
+	filter.Limit = t.taskFindFlags.limit
 
 	var tasks []*influxdb.Task
 
-	if taskFindFlags.id != "" {
-		id, err := influxdb.IDFromString(taskFindFlags.id)
+	if t.taskID != "" {
+		id, err := influxdb.IDFromString(t.taskID)
 		if err != nil {
 			return err
 		}
 
-		task, err := s.FindTaskByID(context.Background(), *id)
+		task, err := tskSvc.FindTaskByID(context.Background(), *id)
 		if err != nil {
 			return err
 		}
 
 		tasks = append(tasks, task)
 	} else {
-		tasks, _, err = s.FindTasks(context.Background(), filter)
+		tasks, _, err = tskSvc.FindTasks(context.Background(), filter)
 		if err != nil {
 			return err
 		}
 	}
 
-	return printTasks(
-		cmd.OutOrStdout(),
-		taskPrintOpts{
-			hideHeaders: taskPrintFlags.hideHeaders,
-			json:        taskPrintFlags.json,
-			tasks:       tasks,
-		},
-	)
+	return t.printTasks(taskPrintOpts{tasks: tasks})
 }
 
-var taskUpdateFlags struct {
-	id     string
+type taskRerunFailedFlags struct {
+	before string
+	after  string
+}
+
+func (t *cmdTaskBuilder) taskRerunFailedCmd() *cobra.Command {
+	cmd := t.opts.newCmd("rerun_failed", t.taskRerunFailedF, true)
+	cmd.Short = "Find and Rerun failed runs/tasks"
+	cmd.Aliases = []string{"rrf"}
+
+	t.org.register(t.opts.viper, cmd, false)
+	t.globalFlags.registerFlags(t.opts.viper, cmd)
+	registerPrintOptions(t.opts.viper, cmd, &t.taskPrintFlags.hideHeaders, &t.taskPrintFlags.json)
+	cmd.Flags().StringVarP(&t.taskID, "id", "i", "", "task ID")
+	cmd.Flags().StringVar(&t.taskRerunFailedFlags.before, "before", "", "before interval")
+	cmd.Flags().StringVar(&t.taskRerunFailedFlags.after, "after", "", "after interval")
+
+	return cmd
+}
+
+func (t *cmdTaskBuilder) taskRerunFailedF(*cobra.Command, []string) error {
+	if err := t.org.validOrgFlags(&flags); err != nil {
+		return err
+	}
+
+	tskSvc, _, err := t.svcFn()
+	if err != nil {
+		return err
+	}
+
+	taskIDPresent := t.taskID == ""
+
+	/*
+		If no TaskID is given, use TaskFilter to get all Tasks, search for failed runs in each, then re run failed
+		If TaskID given, use RunFilter to search for failed runs then re run
+	*/
```

Comment:
Instead of the couple branching if-elses, could you make 2 new helper functions (one for taskID present & one for no taskID, both returning `[]*influxdb.Run`) and use those here? There will be a couple lines of duplication, but I think it'll be cleaner overall.